pages/exercises/[lessonSlug].tsx

```diff
@@ -70,31 +71,34 @@ const Exercises: React.FC<QueryDataProps<GetExercisesQuery>> = ({
       explanation: exercise.explanation || '',
       userAnswer: userAnswers[exercise.id] ?? null
     }))
-
-  const exercise = currentExercises[exerciseIndex]
+    .filter(
+      exercise => !hideAnswered || exercise.userAnswer !== exercise.answer
+    )
 
   return (
     <Layout title={currentLesson.title}>
-      {exercise ? (
+      {solvingExercise ? (
         <Exercise
-          key={exerciseIndex}
```

Comment:
I removed `key` here because we want the `Exercise` component to not rerender since the `Exercise` component now holds more state which we don't want to reset by rerendering.

Another way I could have implemented this would be to keep `userAnswers` and `localUserAnswers` at the same level here and then only pass down `localUserAnswers` to the `Exercise` component.

Both ways work, but I chose to keep `localUserAnswers` at the `Exercise` component level.